src/mame/atlus/cave.cpp

```diff
@@ -1756,6 +1823,27 @@ static INPUT_PORTS_START( paccarn )
 	PORT_BIT( 0x04, IP_ACTIVE_HIGH, IPT_BUTTON5 ) PORT_NAME( "Bet 12" )
 INPUT_PORTS_END
 
+static INPUT_PORTS_START( jumbogod )
+        PORT_START("IN0")
+        PORT_BIT( 0x01, IP_ACTIVE_LOW,  IPT_UNKNOWN )
+        PORT_BIT( 0x02, IP_ACTIVE_LOW,  IPT_COIN1 ) PORT_IMPULSE(10) // credits (impulse needed to coin up reliably)
+        PORT_BIT( 0x04, IP_ACTIVE_LOW,  IPT_UNKNOWN )
+        PORT_BIT( 0x08, IP_ACTIVE_HIGH, IPT_CUSTOM ) PORT_READ_LINE_DEVICE_MEMBER("eeprom", eeprom_serial_93cxx_device, do_read)
+        PORT_BIT( 0x10, IP_ACTIVE_LOW,  IPT_BUTTON1 ) PORT_NAME( "Tai Atari (Body Slam)" )
+        PORT_BIT( 0x20, IP_ACTIVE_LOW,  IPT_BUTTON3 ) PORT_NAME( "Shippo Kougeki (Tail Attack)")  // not a mistake; it is #3 by the game's test menu.
+        PORT_BIT( 0xC0, IP_ACTIVE_LOW,  IPT_UNKNOWN )
+
+        PORT_START("IN1")
+        PORT_SERVICE( 0x01, IP_ACTIVE_LOW ) // must stay on during service mode
+        PORT_BIT( 0x06, IP_ACTIVE_LOW, IPT_UNKNOWN )
+        PORT_CONFNAME( 0x08, 0x08, "Self Test" )
+        PORT_CONFSETTING(    0x08, DEF_STR( Off ) )
+        PORT_CONFSETTING(    0x00, DEF_STR( On ) )
+        PORT_BIT( 0x10, IP_ACTIVE_LOW, IPT_BUTTON2   ) PORT_NAME( "Nissen Kougeki (Atomic Breath)" )
+        PORT_BIT( 0x20, IP_ACTIVE_LOW, IPT_BUTTON4   ) PORT_NAME( "Nissen Kougeki (Atomic Breath)" )
+        PORT_BIT( 0xC0, IP_ACTIVE_LOW, IPT_UNKNOWN )
+INPUT_PORTS_END 
```

Comment:
It’s going to confuse users if there are two identically named buttons.

Also, you should use `PORT_SERVICE` for the test mode switch.

Reply:
In this case, PORT_SERVICE is already being used for the operator test menu in line with how it seems to be used for many titles. The game has a separate diagnostic self test mode, operated by another switch. Do you mean that they should be switched? In the current configuration, striking F2 brings up the operator menu with settings, which is the behavior I expected to see.

Reply:
Oh, sorry, I missed that `PORT_SERVICE` is already being used.  Can you use another one of the service controls?  Is `IPT_SERVICE1` available?  For example something like:
```
        PORT_BIT( 0x08, IP_ACTIVE_LOW, IPT_SERVICE1 ) PORT_TOGGLE PORT_NAME( "Self Test" )
```